### PR TITLE
PAT-2098 Convert to use generic-service

### PR DESCRIPTION
**Pull request by steverendell**
Add features needed by pathfinder, cat tool etc

### CODE REVIEW

File: charts/generic-service/templates/_envs.tpl [hunk 1]
```diff
@@ -3,23 +3,22 @@
 Environment variables for web and worker containers
 */}}
 {{- define "deployment.envs" -}}
-{{- $appName := include "generic-service.name" . -}}
```
Comment (steverendell) [inline charts/generic-service/templates/_envs.tpl:6]: $appName is not used

File: charts/generic-service/templates/_envs.tpl [hunk 2]
```diff
@@ -3,23 +3,22 @@
...
   {{- range $key, $val := $envs }}
   - name: {{ $key }}
     valueFrom:
       secretKeyRef:
-        key: {{ $val }}
-        name: {{ $secret }}
-  {{- end }}
+        key: {{ trimSuffix "?" $val }}
+        name: {{ $secret }}{{ if hasSuffix "?" $val }}
+        optional: true{{ end }}  {{- end }}
```
Comment (steverendell) [inline charts/generic-service/templates/_envs.tpl:15]: allow for secret keys that may not exist

File: charts/generic-service/templates/ingress.yaml [hunk 3]
```diff
@@ -19,21 +19,7 @@ metadata:
...
-      SecRuleUpdateActionById 949110 "t:none,deny,status:406,logdata:%{SERVER_NAME}"
-      SecRuleUpdateActionById 959100 "t:none,deny,status:406,logdata:%{SERVER_NAME}"
-      SecAction \
-        "id:900000,\
-         phase:1,\
-         nolog,\
-         pass,\
-         t:none,\
-         setvar:tx.paranoia_level=2"
+{{ .Values.ingress.modsecurity_snippet | indent 6 }}
```
Comment (steverendell) [inline charts/generic-service/templates/ingress.yaml:22]: make modsecurity-snippet configurable

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+/.idea/
```
Comment (petergphillips) [inline .gitignore:1]: normally better to add a global gitignore then you don't need to do this to all projects that you touch :grin:

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**